programs/mpl-core/src/processor/create.rs

```diff
@@ -146,47 +149,87 @@ pub(crate) fn process_create<'a>(
+                    &new_asset,
                     ctx.accounts.asset,
                     ctx.accounts.payer,
                     ctx.accounts.system_program,
                 )?;
+                for plugin in &plugins {
+                    if PluginType::check_create(&PluginType::from(&plugin.plugin))
+                        != CheckResult::None
+                    {
+                        let validation_ctx = PluginValidationContext {
+                            self_authority: &plugin.authority.unwrap_or(plugin.plugin.manager()),
+                            authority_info: authority,
+                            resolved_authorities: None,
+                            new_owner: None,
+                            target_plugin: None,
+                        };
+                        match Plugin::validate_create(&plugin.plugin, &validation_ctx)? {
+                            ValidationResult::Rejected => approved = false,
+                            ValidationResult::ForceApproved => force_approved = true,
+                            _ => (),
+                        };
+                    }
+                    initialize_plugin::<AssetV1>(
+                        &plugin.plugin,
+                        &plugin.authority.unwrap_or(plugin.plugin.manager()),
+                        &mut plugin_header,
+                        &mut plugin_registry,
+                        ctx.accounts.asset,
+                        ctx.accounts.payer,
+                        ctx.accounts.system_program,
+                    )?;
+                }
             }
+        }
 
-            if !(approved || force_approved) {
-                return Err(MplCoreError::InvalidAuthority.into());
+        if let Some(plugins) = args.external_plugins {
+            if !plugins.is_empty() {
+                let (_, mut plugin_header, mut plugin_registry) = create_meta_idempotent::<AssetV1>(
+                    ctx.accounts.asset,
+                    ctx.accounts.payer,
+                    ctx.accounts.system_program,
+                )?;
+                for plugin_init_info in &plugins {
+                    if ExternalPlugin::check_create(plugin_init_info) != ExternalCheckResult::none()
+                    {
+                        let validation_ctx = PluginValidationContext {
+                            // External plugins are always managed by the update authority.
+                            self_authority: &Authority::UpdateAuthority,
+                            authority_info: authority,
+                            resolved_authorities: None,
+                            new_owner: None,
+                            target_plugin: None,
+                        };
+                        match ExternalPlugin::validate_create(plugin_init_info, &validation_ctx)? {
+                            ValidationResult::Rejected => approved = false,
+                            ValidationResult::ForceApproved => force_approved = true,
```

Comment:
This is what I was thinking when I asked the other question (but which was on the wrong function): i.e. can  `ExternalPlugin::validate_create` ever return `ForceApproved`?

Reply:
Talked offline. It probably isn't something we want to allow so going to remove.